adhocteam/Head-Start-TTADP
# Objectives/Goals are persisted

**Description of change**
 * When a report is saved the goals and objectives from the report are saved to the DB, updated or created if needed
 * An ActivityReportObjective record is created for every objective associating the report to objective
 * Objectives that are no longer used in the report are removed from the DB, along with the ActivityReportObjective
 * If the goal hasn't been used on a previous report and is removed from the current report it is also removed
 * When a report is approved a GrantGoal record is created for every goal/grant pair

**How to test**
1) Pull down, run migrations
2) Create a new report, select one or more grants and complete the report. Add a few goals/objectives to the report
3) Approve the report
4) Start another new report, select a few of the same grants as the first report. Make sure the goals you previously created are select-able in the goal view (The first report must be approved for the goals to show up in the new report!)

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/287

**Checklist**

    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
